### PR TITLE
Surgery success to speed conversion, failure removed

### PR DESCRIPTION
**Pull request by jdawg1290**
## About The Pull Request
Changes surfaces and tools to affect surgery speed instead of success rates. Surgery can now be performed on floors at a steep penalty. Surgeries cannot be failed.

## Why It's Good For The Game
Should encourage use of field surgery while still keeping proper surgery tables and tools as the superior choice.

## Changelog
:cl: Jdawg1290
balance: Surgery surface and tools now affects speed instead of success rate. Surgeries cannot be failed.
/:cl:

### CODE REVIEW

**Comment by spookydonut** [comment]
`propability`

**Comment by jdawg1290** [comment]
might as well while i'm in there

**Comment by Mickyan** [comment]
floor surgery was already a thing, it used the default probability of 0.5, so you didn't have to change the default value and add that extra `else` at the bottom but I don't think it really matters much

**Comment by jdawg1290** [comment]
The else for speed makes sure that nontable/bed surgery is slowed, otherwise you do surgery as fast on the floor as an operating table.

**Comment by Mickyan** [comment]
look at how probability worked before, the default probability is at 0.5 (floor surgery) and it only gets increased if it can locate a table on the turf

**Comment by coiax** [comment]
I'm surprised this isn't a more contentious change, frankly.

File: code/modules/surgery/surgery.dm [hunk 1]
```diff
@@ -109,19 +109,25 @@
 	qdel(src)
 
 /datum/surgery/proc/get_propability_multiplier()
-	var/propability = 0.5
+	var/probability = 1.0
+
+	return probability + success_multiplier
+
+/datum/surgery/proc/get_speed_multiplier()
+	var/speed = 1.0
```
Comment (ExcessiveUseOfCobblestone) [inline code/modules/surgery/surgery.dm]: you should just set this to 0.25 and remove 128/129

**Comment by ExcessiveUseOfCobblestone** [comment]
this is a really hacky way to do this lol

**Comment by jdawg1290** [comment]
I could move the speed and probability calculation into the surgery step fairly easily, in fact the get_probability is obsolete now, since the surgery_step/initiate proc gets a reference to the surgery and its success var.

**Comment by ExcessiveUseOfCobblestone** [comment]
You should remove probability proper as well as failing mechanics. Like you said, they're obsolete.

**Comment by jdawg1290** [comment]
Would you prefer i go ahead and convert tool failure to delay too?

**Comment by optimumtact** [comment]
@coiax why would it be contentious, the previous mechanic punished you for failing, even though there was no actual skill component involved.

This is a better way to do it, because you're still encouraged to use the right gear and tables, but you're not punished for something that is outside of your skill to counter

**Comment by jdawg1290** [comment]
I seem to have broken git :/ but i have done some testing and things are working as intended, floor dissection with a kitchen knife takes ~3 minutes 45 seconds, but always succeeds. nothing can be failed anymore.

**Comment by jdawg1290** [comment]
Closing to cleanup branch

**Comment by ExcessiveUseOfCobblestone** [comment]
how promising

**Pull request closed without merging**